Add pwndbg commands list filtering

disconnect3d opened the pull request:
Hello,

I have added a possibility to filter the list of commands provided with `pwndbg`, simply by passing an argument to it:

```
pwndbg> pwndbg entry
entry                Set a breakpoint at the first instruction executed in
entry_point          GDBINIT compatibility alias to print the entry point.
pwndbg> 
```

Currently, this is done by simple lowercase string comparison. Imho we don't really need regexes here.

zachriggle commented:
I would prefer that all new commands be of the type `ArgparsedCommand`.  Would you mind modifying your pull request to use this?

Thanks!

disconnect3d commented:
I've changed the commands in misc.py to use `ArgparsedCommands`.

* Should I also split them into different files?
* Is the `type=int` proper for `errno` and `distance` commands arguments? Those arguments were `gdb.Value` before this change. Passing `type=gdb.Value` to `parser.add_argument` doesn't work:

  ```
  pwndbg> distance 2 0x33
  Traceback (most recent call last):
    File "/home/dc/pwndbg/pwndbg/commands/__init__.py", line 54, in __call__
      return self.function(*args, **kwargs)
    File "/home/dc/pwndbg/pwndbg/commands/__init__.py", line 159, in _ArgparsedCommand
      return function(**vars(args))
    File "/home/dc/pwndbg/pwndbg/commands/misc.py", line 71, in distance
      a = int(a) & _arch.ptrmask
  gdb.error: Cannot convert value to long.
  ```

zachriggle commented:
Would you mind rebasing against `master` and removing the changes to the `distance` command?

Thanks!

disconnect3d commented:
Done.

Currently, as the 2nd commit states, only the `pwndbg` and `errno` commands use argparse in `misc.py` file.

By the way, if you think that we need tests for those commands, feel free to create a task for it and assign me to it.

I am glad to help in my free time ;).

zachriggle commented:
Reaffirming I haven't forgotten this, just on holiday right now :-).
